cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.linkedin.kafka.cruisecontrol.async.OperationFuture;
+import com.linkedin.kafka.cruisecontrol.common.KafkaCruiseControlThreadFactory;
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.apache.kafka.common.utils.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * {@link UserTaskManager} keeps track of long-running request.
+ *
+ * {@link HttpServletRequest} can execute for long durations. The servlet submits the asynchronous tasks and returns the
+ * progress of the operation instead of blocking for the operation to complete. {@link UserTaskManager} maintains the
```

Comment:
Does it return (1) immediately, or (2) blocked for `X` seconds and returns the progress if the request processing has not been completed  within `X` seconds? Would it help to update the JavaDoc to specify it?

Reply:
@efeg I think that could be specific to the endpoint. For example, tomorrow we could add a new endpoint that may return instantaneously instead of waiting for few seconds before returning the current progress. From the `UserTaskManager` it doesn't matter as long as the `UserTaskInfo` contains the Future to check if the action was completed or get the progress.